src/main/kotlin/com/terraformation/backend/file/model/FileMetadata.kt

```diff
@@ -1,33 +1,54 @@
 package com.terraformation.backend.file.model
 
 import com.terraformation.backend.db.asNonNullable
+import com.terraformation.backend.db.default_schema.FileId
 import com.terraformation.backend.db.default_schema.tables.pojos.FilesRow
 import com.terraformation.backend.db.default_schema.tables.references.FILES
+import java.net.URI
 import org.jooq.Record
 import org.springframework.http.MediaType
 
-data class FileMetadata(
-    val filename: String,
+data class FileMetadata<I : FileId?, U : URI?>(
```

Comment:
The optional types to support Nothing (new file usecase)?

Reply:
Yeah, exactly. `Nothing?` is a subtype of all nullable types, but there are no valid values of `Nothing`, so the only valid value of `Nothing?` is null. It pretty much exists for this exact "value must be null" use case.